R/target_market_share.R

```diff
@@ -316,15 +315,72 @@ target_market_share <- function(data,
   data <- data %>%
     pivot_wider2()
 
+  data <- data %>%
+    left_join(green_or_brown, by = c("sector", "technology")) %>%
+    left_join(tmsr_or_smsp, by = "green_or_brown") %>%
+    rename(target_name = .data$which_metric) %>%
+    select(-.data$green_or_brown)
+
   corporate_economy <- calculate_ald_benchmark(ald, region_isos, by_company)
 
   relevant_sectors <- unique(data$sector)
 
   relevant_corporate_economy <- corporate_economy %>%
     filter(.data$sector %in% relevant_sectors)
 
+  relevant_corporate_economy <- relevant_corporate_economy %>%
+    left_join(green_or_brown, by = c("sector", "technology")) %>%
+    left_join(tmsr_or_smsp, by = "green_or_brown") %>%
+    rename(target_name = .data$which_metric) %>%
+    select(-.data$green_or_brown)
+
+  data <- data %>%
+    dplyr::bind_rows(relevant_corporate_economy)
+
+  percent_by_sector_groups <- maybe_add_name_ald(
+    c("sector", "region", "scenario_source", "metric"),
+    by_company
+  )
```

Comment:
very hard to guess what this `maybe_add_name_ald()` function does

Reply:
good point, I tried to clarify this in #384